src/shogun/machine/gp/Inference.cpp

```diff
@@ -92,15 +92,8 @@ void Inference::init()
 	SG_ADD(&m_features, "features", "Features");
 	SG_ADD(&m_gradient_update, "gradient_update", "Whether gradients are updated");
 
-
-	m_kernel=NULL;
-	m_model=NULL;
-	m_labels=NULL;
-	m_features=NULL;
-	m_mean=NULL;
 	m_log_scale=0.0;
 	m_gradient_update=false;
```

Comment:
since you are cleaning this up, could you move the default values to the class member declaration?